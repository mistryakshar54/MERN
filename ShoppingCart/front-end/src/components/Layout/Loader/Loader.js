import React, { Component } from 'react';
import {connect} from 'react-redux';
import ErrorComponent from '../Error/Error';
import Spinner from "react-bootstrap/Spinner";

import './Loader.scss';
export class Loader extends Component{
    render(){
        let { dataLoaded } = this.props;
        console.log( this.props, this.props.apiData );
        if( this.props.apiData.loadingState === true ){
            return (
              <div className="Spinner">
                <Spinner animation="border" size="lg" role="status">
                  <span className="sr-only">Loading...</span>
                </Spinner>
              </div>
            );
        }
        else{
            if (this.props.apiData.apiStatus === 200 && dataLoaded === true) {
              return this.props.children;
            } else {
              return <ErrorComponent />;
            }
        }
    }
}
const mapStateToProps = state => {
  return {
    apiData: state.CoreReducer
  };
};
export default connect(
  mapStateToProps
)(Loader); 
