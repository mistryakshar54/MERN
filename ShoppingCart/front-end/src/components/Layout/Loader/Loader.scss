.Spinner
{
    width:100%;
    align-content: center;
    color: #dc3545;
    padding: 3%;

    .spinner-border {
    width: 4rem;
    height: 4rem;
    border-left-color: transparent;
    border-right-color: transparent;
    }
}