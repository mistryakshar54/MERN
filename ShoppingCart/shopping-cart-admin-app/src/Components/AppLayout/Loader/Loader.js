import React from 'react';
import { useSelector , shallowEqual } from "react-redux";
const Loader = ( props ) => {
    const coreData =  useSelector( state => state.CoreReducer ,  shallowEqual);
    debugger;
    console.log("coreData", coreData.loadingState);
   
        if(coreData.loadingState){
            return <div>Loaded!</div>
        }
        else{
            return props.children;
        }
        
}
export default Loader;
