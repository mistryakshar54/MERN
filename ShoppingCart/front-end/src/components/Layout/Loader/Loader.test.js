import React from "react";
import { mount } from "enzyme";
import {Loader} from "./Loader";
import "../../../setupTests";
import configureMockStore from "redux-mock-store";
import expect from 'expect';
const mockStore = configureMockStore();

let props = {
  dataLoaded: false,
  apiData: {
    loadingState: true,
    apiStatus : 200
  }
};

function createComponent(props ) {
  return mount(<Loader {...props} />);
}
function createComponentWithChildren(props) {
  return mount(
    <Loader {...props}>
      <div id="dataToLoad">Data Loaded!</div>
    </Loader>
  );
}

describe("Loader Component Test", () => {
    it("Renders the component correctly", () => {
        
        const wrapper = createComponent(props);
        expect(wrapper.debug()).toMatchSnapshot();
    });

    it("Renders Loader if the api is still loading", () => {
        props.apiData.loadingState = true;
        const wrapper = createComponent(props);
        expect(wrapper.exists(".Spinner")).toBe(true);
        expect(wrapper.debug()).toMatchSnapshot();   
    });

    it("Renders Children if the results are fetched correctly", () => {
      props.apiData.loadingState = false;
      props.dataLoaded = true;
      const wrapper = createComponentWithChildren(props);
      expect(wrapper.find("#dataToLoad").text()).toEqual("Data Loaded!");
      expect(wrapper.debug()).toMatchSnapshot();
    });
});
