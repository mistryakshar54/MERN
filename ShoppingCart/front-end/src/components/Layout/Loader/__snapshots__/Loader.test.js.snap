// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Component Test Renders Children if the results are fetched correctly 1`] = `ReactWrapper {}`;

exports[`Loader Component Test Renders Loader if the api is still loading 1`] = `ReactWrapper {}`;

exports[`Loader Component Test Renders the component correctly 1`] = `ReactWrapper {}`;
